### PR TITLE
add Task5

### PR DESCRIPTION
**Pull request by TsirulKate**

### CODE REVIEW

File: src/servlet/MyServlet.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+    public class MyServlet extends HttpServlet {
+        @Override
+        protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+            try(OutputStreamWriter writer=new OutputStreamWriter(resp.getOutputStream())){
```
Comment (smiakchilo) [inline src/servlet/MyServlet.java]: (cosmetic) Space between `try` and `(` is needed. Spaceas around `=` are needed throughout the project. These are some codestyle standards

File: src/servlet/MyServlet.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                writer.write("Java Web Application");
```
Comment (smiakchilo) [inline src/servlet/MyServlet.java]: Yes, this is one way to get the string printed on screen - by using a servlet. But to be precise, the task required to do this via a static page 
image

File: src/servlet/ServletName.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import java.io.OutputStreamWriter;
+import java.util.Map;
+
+public class ServletName extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException{
+        try(OutputStreamWriter writer=new OutputStreamWriter(resp.getOutputStream())){
+            Map<String,String[]> parameters = req.getParameterMap();
+            String name=parameters.getOrDefault("name", new String[]{"Kate"})[0];
+            if(req.getAttribute("name")!=null){
```
Comment (smiakchilo) [inline src/servlet/ServletName.java]: You have probably copied these lines from our in-class exersise. There it played some (however not pretty) sense - to "override" a parameter map value with a falue ssigned in filter. But yoyu don't have no filter assigning values to requests around here. So the lines 17-19 are in vain here

File: src/servlet/ServletPage.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+public class ServletPage extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        String path = "/WEB-INF/page.html";
```
Comment (smiakchilo) [inline src/servlet/ServletPage.java]: Are you sure you have such path in your code base? Couldn't figure out where the `WEB-INF` folder at all )) 
image
Or, am I looking the wrong way?)

File: src/servlet/ServletStatus.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+
+public class ServletStatus extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        try(OutputStreamWriter writer=new OutputStreamWriter(resp.getOutputStream())){
```
Comment (smiakchilo) [inline src/servlet/ServletStatus.java]: Please take care about the spaces)

**Review by smiakchilo — COMMENTED** [review]
To be honest, could not quite finish the review due to some disruption in your codebase
image
You've got both `.:mockups` and `mockups` folders in your `Task5` branch. You need only one, and properly named (colon is a banned filename char in Windows at least).

You've got two sets of servlets, one in `/src/servlet`, the other in `Task5/src/servlet`.

And a mystery of `WEB-INF` whereabouts ))

Please definitely fix it and make sure project is well run locally, and all the web pages are shown as they must.

File: Task5/src/FilterTime.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+        long start = System.nanoTime();
+        filterChain.doFilter(servletRequest, servletResponse);
+        long end = System.nanoTime();
+        long time = Math.round(end - start);
```
Comment (smiakchilo) [inline Task5/src/FilterTime.java]: What is being rounded here?) `Math.round()` is for retrieving an integer value from a floating-point value, but that's not the case

File: src/servlet/ServletPage.java (hunk 4 above)
Comment (TsirulKate) [reply src/servlet/ServletPage.java]: The assignment was prescribed that index.html should be in the WEB-INF folder.

Comment (smiakchilo) [reply src/servlet/ServletPage.java]: Yes, sure! I just could not find no WEB-INF folder in your branch! Did I miss it?

Comment (TsirulKate) [reply src/servlet/ServletPage.java]: I added it in new commit

**Pull request merged**